Decorator `attrib` unsupperoted where `attr.ib` is

While pylint supports `attr` classes with `attr.ib` attributes, it does not support the same with `attr.attrib`, which is an alias preferred by some.

### Steps to reproduce
1. `pylint attrib_test.py` with:
```python
import attr
from attr import attrib, attrs

@attr.s
class D1:
    d = attr.ib()

@attrs
class D2:
    d = attrib()

@attr.s
class OK:
    x1 = attr.ib(type=D1)
    x2 = attr.ib(type=D2)

    def test(self):
        return (self.x1.d, self.x2.d)

@attrs
class Err:
    x1 = attrib(type=D1)
    x2 = attrib(type=D2)

    def test(self):
        return (self.x1.d, self.x2.d)
```
### Current behavior

Class `OK` reports no errors while `Err` reports:
```
[... snip docstring and too-few-methods lints]
tst.py:26:16: E1101: Instance of '_CountingAttr' has no 'd' member (no-member)
tst.py:26:27: E1101: Instance of '_CountingAttr' has no 'd' member (no-member)
[... snip docstring and too-few-methods lints]
```
Note the problem is not in how classes `D1`, `D2` were declared (`attr.ib` or `attrib`).

### Expected behavior

class `Err` should also return no errors.

### ``python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"`` output

2.1.0

--- a/ChangeLog
+++ b/ChangeLog
@@ -14,6 +14,10 @@ Release Date: TBA
 
   Close PyCQA/pylint#2534
 
+* Extend detection of data classes defined with attr
+
+  Close #628
+
 
 What's New in astroid 2.1.0?
 ============================

--- a/astroid/brain/brain_attrs.py
+++ b/astroid/brain/brain_attrs.py
@@ -11,12 +11,11 @@ import astroid
 from astroid import MANAGER
 
 
-ATTR_IB = "attr.ib"
+ATTRIB_NAMES = frozenset(("attr.ib", "attrib", "attr.attrib"))
+ATTRS_NAMES = frozenset(("attr.s", "attrs", "attr.attrs", "attr.attributes"))
 
 
-def is_decorated_with_attrs(
-    node, decorator_names=("attr.s", "attr.attrs", "attr.attributes")
-):
+def is_decorated_with_attrs(node, decorator_names=ATTRS_NAMES):
     """Return True if a decorated node has
     an attr decorator applied."""
     if not node.decorators:
@@ -41,7 +40,7 @@ def attr_attributes_transform(node):
         if not isinstance(cdefbodynode, astroid.Assign):
             continue
         if isinstance(cdefbodynode.value, astroid.Call):
-            if cdefbodynode.value.func.as_string() != ATTR_IB:
+            if cdefbodynode.value.func.as_string() not in ATTRIB_NAMES:
                 continue
         else:
             continue

--- a/astroid/tests/unittest_brain.py
+++ b/astroid/tests/unittest_brain.py
@@ -1030,6 +1030,7 @@ class AttrsTest(unittest.TestCase):
         module = astroid.parse(
             """
         import attr
+        from attr import attrs, attrib
 
         @attr.s
         class Foo:
@@ -1045,10 +1046,24 @@ class AttrsTest(unittest.TestCase):
 
         g = Bar()
         g.d['answer'] = 42
+
+        @attrs
+        class Bah:
+            d = attrib(attr.Factory(dict))
+
+        h = Bah()
+        h.d['answer'] = 42
+
+        @attr.attrs
+        class Bai:
+            d = attr.attrib(attr.Factory(dict))
+
+        i = Bai()
+        i.d['answer'] = 42
         """
         )
 
-        for name in ("f", "g"):
+        for name in ("f", "g", "h", "i"):
             should_be_unknown = next(module.getattr(name)[0].infer()).getattr("d")[0]
             self.assertIsInstance(should_be_unknown, astroid.Unknown)
 

